### ISSUE TITLE
2D map not appear

### ISSUE THREAD

**Issue by darissa** [verlab/3d-object-semantic-mapping#5]
Hi Verlab,

I am new to ROS. I have follow the step by step that you provided. However, on rviz, only tf odom appear. 2D map not generated. I am using ubuntu 18.04, ros Melodic, cuda 10.2.

**Comment by renatojmsdh** [comment]
Hello @darissa , sorry for the late reply. I hope you succeed running and trying the code. The packages and launches were developed and mostly tested with ROS Kinetic. We provided some portability instructions for the turtlebot2 and TF's for Melodic here https://github.com/verlab/3DSemanticMapping_JINT_2020/blob/master/docs/INSTALL_MELODIC.md. Maybe the map/turtlebot2 TF can be the reason why the 2D occupancy map don't show up. 
We are currently refactoring this code and creating a new branch to be used with ROS Noetic (with a simpler running scripts) likely this week. We keep you posted as soon as we have created the noetic branch. Best

**Comment by darissa** [comment]
Yes, I have followed the installation guide for melodic. Already surrender =D. But went back to your page as I still cannot solve my problem. I have same objective like yours. Just need to replace the config & weight file as I want to detect Tree. Plus I need display Tree in 2D map just like Doors in your 2D map. Please help if you have any idea or link or somewhere that I can refer. Thank you.

**Comment by darissa** [comment]
@renatojmsdh Btw, what version opencv do you used? I got error during compile:

1. error: ‘cvResizeWindow’ was not declared in this scope
       cvResizeWindow("YOLO V3", 640, 480);

2. error: no matching function for call to ‘_IplImage::_IplImage(cv::Mat&)’

I am using opencv-4.5.3.

**Comment by renatojmsdh** [comment]
Hello @darissa, 

The opencv version was the one used in darknet_ros package: 2.4.9.1 (it also the one provided by default in the software manager of ubuntu xenial). And there have been indeed strong changes that happened from opencv 2.x to opencv 3.x and to opencv 4.x. I think it is unlike you will be able to install this version of darknet with opencv4.x because of c++11 support required now on. If you can use opencv 3.4 I think these modifications can work for compiling darknet_ros: 

Please replace the following lines in darknet_ros/darknet_ros/src/YoloObjectDetector.cpp for fixing opencv renamed functions including resizeWindow, namedWindow, ... since opencv 3.x:

- line 431:   ```shell int c = cv::waitKey(waitKeyDelay_);```
- lines 522 - 528:
```shell
  if (!demoPrefix_ && viewImage_) {
	  cv::namedWindow("YOLO V3", cv::WINDOW_NORMAL);
    if (fullScreen_) {
	    cv::setWindowProperty("YOLO V3", cv::WND_PROP_FULLSCREEN, cv::WINDOW_FULLSCREEN);
    } else {
	    cv::moveWindow("YOLO V3", 0, 0);
	    cv::resizeWindow("YOLO V3", 640, 480);
```
- lines 562 - 567: (https://stackoverflow.com/questions/4664187/converting-cvmat-to-iplimage)
```shell
IplImage* YoloObjectDetector::getIplImage()
{
  boost::shared_lock<boost::shared_mutex> lock(mutexImageCallback_);
  IplImage ROS_img = cvIplImage(camImageCopy_);
  return &ROS_img;
}
```
best

**Comment by KalanaRatnayake** [comment]
Were you able to finish the porting process for noetic? I'm attempting to set this on melodic. map frame seems to be missing from tf

**Comment by deavn2236** [comment]
> 你好@darissa， 这么晚才回复很抱歉。我希望您成功运行并尝试代码。软件包和发布是使用 ROS Kinetic 开发和测试的。[我们在此处https://github.com/verlab/3DSemanticMapping_JINT_2020/blob/master/docs/INSTALL_MELODIC.md](https://github.com/verlab/3DSemanticMapping_JINT_2020/blob/master/docs/INSTALL_MELODIC.md)提供了一些针对 turtlebot2 和 TF 的可移植性说明。也许 map/turtlebot2 TF 可能是 2D 占用图未显示的原因。 我们目前正在重构这段代码，并可能在本周创建一个与 ROS Noetic（具有更简单的运行脚本）一起使用的新分支。一旦我们创建了 noetic 分支，我们就会及时通知您。最好的

I have the same problem,Were you able to finish the porting process for noetic?

**Comment by Xuefei98** [comment]
Same question +1, thank you!

**Comment by darissa** [comment]
sorry guys as I was not able to solve this.
